staking-cli/src/delegation.rs

```diff
@@ -5,7 +5,22 @@ use alloy::{
     transports::Transport,
 };
 use anyhow::Result;
-use contract_bindings_alloy::staketable::StakeTable::StakeTableInstance;
+use contract_bindings_alloy::{
+    esptoken::EspToken::EspTokenInstance, staketable::StakeTable::StakeTableInstance,
+};
+
+pub async fn approve<P: Provider<T>, T: Transport + Clone>(
+    token: EspTokenInstance<T, P>,
+    stake_table_address: Address,
+    amount: U256,
+) -> Result<TransactionReceipt> {
```

Comment:
i'll rectify this API as I have suggested on Zulip. 
accept `provider, address` instead of accepting an instance.